test/uri/LongUriSerializerTest.cpp

```diff
@@ -0,0 +1,790 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <string>
+#include <cgreen/cgreen.h>
+#include <uprotocol-cpp/uri/serializer/LongUriSerializer.h>
+#include <uprotocol-cpp/uri/datamodel/UUri.h>
+#include <uprotocol-cpp/uri/datamodel/UAuthority.h>
+#include <uprotocol-cpp/uri/datamodel/UEntity.h>
+#include <uprotocol-cpp/uri/datamodel/UResource.h>
+
+using namespace cgreen;
+using namespace uprotocol::uri;
+
+#define assertTrue(a) assert_true(a)
+#define assertEquals(a, b) assert_true(b == a)
+#define assertFalse(a) assert_false(a)
+
+Describe(LongUriSerializer);
+
+BeforeEach(LongUriSerializer) {
```

Comment:
What are these for? BeforeEach() and AfterEach()?

Reply:
this is part of CGREEN tests , we are in progress of changing to GTEST